### PR TITLE
Fix: false positive in BufferFilesNewStream #456

### LINKED ISSUE #456: Fix Request: false positive in BufferFilesNewStream
IOUtils buffers the InputStream, so no extra buffering needed. OutputStream is however not buffered.
```
void ioUtilsBad(InputStream in, String path) throws IOException {
        OutputStream out = Files.newOutputStream(Paths.get(path)); // bad, not buffered
        IOUtils.copy(in, out);
    }
    void ioUtilsGood1(InputStream in, String path) throws IOException {
        OutputStream out = IOUtils.buffer(Files.newOutputStream(Paths.get(path)));
        IOUtils.copy(in, out);
    }
    void ioUtilsGood2(InputStream in, String path) throws IOException {
        OutputStream out = Files.newOutputStream(Paths.get(path));
        IOUtils.copy(in, IOUtils.buffer(out));
    }
    void ioUtilsGood3(String pathIn, String pathOut) throws IOException {
        InputStream in = Files.newInputStream(Paths.get(pathIn));
        OutputStream out = Files.newOutputStream(Paths.get(pathOut));
        IOUtils.copy(in, IOUtils.buffer(out)); // buffer inputstream
    }

```

### PR DESCRIPTION
**Pull request by jborgers**

### CODE REVIEW

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/BufferFilesNewStream.xml [hunk 1]
```diff
@@ -74,4 +74,34 @@ class Issue273 {
...
+        <expected-linenumbers>7</expected-linenumbers>
+        <code><![CDATA[
+import java.io.*;
+import java.nio.file.*;
+import org.apache.commons.io.IOUtils;
+
+class Issue456 {
+    void ioUtilsBad(InputStream in, String path) throws IOException {
+        OutputStream out = Files.newOutputStream(Paths.get(path)); // bad, not buffered
+        IOUtils.copy(in, out);
```
Comment (stokpop) [inline src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/BufferFilesNewStream.xml:89]: what if out is not File based? E.g. network or in memory?

**Review by stokpop — APPROVED** [review]
One question. Might give false positives for non-file cases?

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/BufferFilesNewStream.xml (hunk 1 above)
Comment (jborgers) [reply src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/BufferFilesNewStream.xml:89]: It is about the Files class, so meant for files not other I/O.

Comment (jborgers) [reply src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/BufferFilesNewStream.xml:89]: IOUtils is for other I/O than just files, yet Files.newXXX is meant for files (and directories)

**Pull request merged**